### PR TITLE
perf: move id: eq fast path handling to node-model so it's shared between query running strategies

### PR DESCRIPTION
**Pull request by pieh**
## Description

It seems we lost our fast path for simple `id.eq` queries and only had path like that for LMDB (experimental) indexes. I moves this to much earlier in pipeline (node-model).

This should mean fast filters WON'T create lookup table for it. Idea here is to limit usage of it where possible, as (right now) those lookup tables cause nodes to be retained in memory and also not needed as we can already grab nodes by id.

## Related Issues

[sc-44551]

### CODE REVIEW

File: packages/gatsby/src/schema/node-model.js [hunk 1]
```diff
@@ -267,6 +267,42 @@ class LocalNodeModel {
...
+
+    // check if we can get node by id and skip
+    // more expensive query pipeline
+    if (
+      typeof query?.filter?.id?.eq !== `undefined` &&
+      Object.keys(query.filter).length === 1 &&
+      Object.keys(query.filter.id).length === 1
+    ) {
+      if (tracer) {
+        runQueryActivity = reporter.phantomActivity(`runQuerySimpleIdEq`, {
```
Comment (wardpeet) [inline packages/gatsby/src/schema/node-model.js:280]: What's the goal of this phantomActivity? To see it in tracing?

**Review by wardpeet — APPROVED** [review]
LGTM! Nice catch!

File: packages/gatsby/src/schema/node-model.js (hunk 1 above)
Comment (pieh) [reply packages/gatsby/src/schema/node-model.js:280]: Yes, just for instrumentation. This code path skips `runQuery` activity, so I replaced it with something so traces make more sense

**Comment by KyleAMathews** [comment]
Woot!

**Pull request merged**